app/app_windows.go

```diff
@@ -73,11 +73,21 @@ func (a *fyneApp) SendNotification(n *fyne.Notification) {
 	title := escapeNotificationString(n.Title)
 	content := escapeNotificationString(n.Content)
 	appID := a.UniqueID()
+	iconFilePath := filepath.Join(os.TempDir(), fmt.Sprintf("fyne-%s.ico", appID))
 	if appID == "" || strings.Index(appID, "missing-id") == 0 {
 		appID = a.Metadata().Name
 	}
 
-	script := fmt.Sprintf(notificationTemplate, title, content, appID)
+	if a.Icon() != nil {
+		if _, err := os.Stat(iconFilePath); os.IsNotExist(err) {
+			err := os.WriteFile(iconFilePath, a.Icon().Content(), 0600)
```

Comment:
yes, it was for that reason that I used os.Stat to check if the icon already existed, because saving the icon every time seemed like bad design/performance. But the idea behind the cachedIconPath is superb. I will implement it that way.